byterun/caml/mlvalues.h

```diff
@@ -277,12 +277,48 @@ CAMLextern void caml_Store_double_val (value,double);
 
 /* Arrays of floating-point numbers. */
 #define Double_array_tag 254
-#define Double_field(v,i) Double_val((value)((double *)(v) + (i)))
-#define Store_double_field(v,i,d) do{ \
+
+/* The [_flat_field] macros are for [floatarray] values and float-only records.
```

Comment:
Somewhere there should be a note observing that record types "obviously of all floats" are still represented using tag `Double_array_tag`.  There may otherwise be a misconception that this optimisation has also been removed.

Reply:
(This has been addressed)